apps/strapi-admin-extensions/src/controllers/import/index.ts

```diff
@@ -0,0 +1,64 @@
+import type { NextFunction, Request, Response } from 'express';
+import fs from 'node:fs';
+import pLimit from 'p-limit';
+import { CREATE_VAC } from '../../queries';
+import { CreateVacResponse } from '../../strapi-product-types';
+import { fetchData, processCsvFile } from '../../utils';
+
+const limit = pLimit(5); // Limit the number of concurrent file uploads
+export const importController = async (req: Request, res: Response, next: NextFunction) => {
+  const type = req.body.type;
+  if (req.file) {
+    const filePath = req.file.path;
+    const requiredColumns = ['vraag', 'antwoord'];
+
+    try {
+      // Process the CSV file and sanitize results
+      const authorizationHeader = req.headers?.authorization || '';
+      const [authType, authToken] = authorizationHeader.split(/\s+/);
+      const tokenAuth = authType === 'Token' ? authToken : authorizationHeader;
+      const graphqlURL = new URL('/graphql', process.env.STRAPI_PRIVATE_URL);
+      const sanitizedResults = await processCsvFile(filePath, requiredColumns);
+      const locale = req.query?.locale || 'nl';
+
+      if (type === 'vac') {
+        // Loop through the sanitized results and create entries one by one
+        const results = await Promise.all(
+          sanitizedResults.map((entry) =>
+            limit(async () => {
+              try {
+                const { data: responseData } = await fetchData<CreateVacResponse>({
+                  url: graphqlURL.href,
+                  query: CREATE_VAC,
+                  variables: { locale, data: entry },
+                  headers: {
+                    Authorization: `Bearer ${tokenAuth}`,
+                  },
+                });
+                return responseData;
+              } catch (error: any) {
+                next(error);
+                // eslint-disable-next-line no-console
+                console.error('Error processing entry:', error);
+                return { error: error.message, entry };
+              }
+            }),
+          ),
+        );
```

Comment:
Zou het niet mogelijk zijn om de Overige Objecten API te gebruiken in plaats van direct met Strapi te connecten? Dan werkt deze importer ook met andere backends.